### PR TITLE
Various fixes around the daemon

### LINKED ISSUE #9: Proper progress calculation
We should implement some proper progress calculation over double threading each task, or whatever is going on right now.

### LINKED ISSUE #11: Don't run tasks on folders that already are part of other tasks
If we have a running task for, let's say, /home but we try to add a new task for, let's say, /home/user, we should not create the task but send back an ALREADY_RUNNING error or something.

### LINKED ISSUE #12: be able to REMOVE not finished tasks

### PR DESCRIPTION
**Pull request by oxygen-consumer**
fixes #9 
fixes #11 
fixes #12

### CODE REVIEW

File: daemon/src/analyzer.c [hunk 1]
```diff
@@ -64,6 +73,10 @@ long long analyzing(const char *path, struct task_details *task)
     long long size = 0;
     DIR *directory = opendir(path);
 
+    double dirs_count = 0;
+    PathStack *stack = (PathStack *)malloc(sizeof(PathStack));
+    initialize_path_stack(stack);
+
     if (directory == NULL)
         return 0;
```
Comment (VoicilaIonut) [inline daemon/src/analyzer.c:82]: can the following while be a function now ?

File: daemon/src/analyzer.c [hunk 2]
```diff
@@ -53,9 +54,17 @@ long long get_size_dir(const char *path, struct task_details *task)
     return size;
 }
 
-long long analyzing(const char *path, struct task_details *task)
+long long analyzing(const char *path, struct task_details *task, double sub_progress)
```
Comment (VoicilaIonut) [inline daemon/src/analyzer.c:57]: please leave some comments about how sub_progress works

**Review by VoicilaIonut — CHANGES_REQUESTED** [review]
I added some small comments.
I dont understand exactly how sub_progress works, please leave some comments in the code.
Thanks for fixing remove!

File: daemon/src/analyzer.c (hunk 1 above)
Comment (oxygen-consumer) [reply daemon/src/analyzer.c:82]: I guess so, I can try to make it a while if you want.

**Review by VoicilaIonut — APPROVED** [review]

**Pull request merged**